zhaquirks/tuya/ts0601_trv.py

```diff
@@ -931,46 +921,288 @@ async def command(
+
+    def __new__(cls, value):
+        """Create a string representation of the attribute."""
+        if isinstance(value, str):
+            return super().__new__(cls, value)
+        else:
+            s = "".join(chr(x) for x in value)[::-1]
+            humanstr = (
+                s[0:4]
+                + "/"
+                + s[4:6]
+                + "/"
+                + s[6:8]
+                + " "
+                + s[8:10]
+                + ":"
+                + s[10:12]
+                + " | "
+                + s[12:16]
+                + "/"
+                + s[16:18]
+                + "/"
+                + s[18:20]
+                + " "
+                + s[20:22]
+                + ":"
+                + s[22:24]
+            )
+
+            return super().__new__(cls, humanstr)
+
+    def serialize(self):
+        """Serialize holiday start stop values into binary values."""
+
+        data = self
+        if not re.match(
+            r"\d\d\d\d/\d\d/\d\d \d\d:\d\d \| \d\d\d\d/\d\d/\d\d \d\d:\d\d", str(data)
+        ):
+            raise ValueError(
+                "Holiday Start Stop must be in the following format:"
+                " 2021/01/01 01:01 | 2021/01/01 01:01"
+            )
+        b = (
+            data[0:4]
+            + data[5:7]
+            + data[8:10]
+            + data[11:13]
+            + data[14:16]
+            + data[19:23]
+            + data[24:26]
+            + data[27:29]
+            + data[30:32]
+            + data[33:35]
+        )
+
+        ret = b[::-1].encode()
+        return ret
+
+
+class ZONNSMARTScheduleSet(t.FixedList, item_type=t.uint8_t, length=31):
+    """Set the schedule for one or multiple days."""
+
+    def __init__(self, value, *args, **kwargs):
+        """Parse the schedule string and serialize it."""
+
+        super().__init__(*args, **kwargs)
+        if value is None:
+            return
+
+        if not isinstance(value, str):
+            raise ValueError("Only string value allowed")  # noqa: TRY004
+
+        periods = value.split()
+        if len(periods) < 2:
+            raise ValueError("At least a day and a period are needed")
+
+        b = []
+        day = periods[0]
+        periods.pop(0)
+        if day == "monday" or day == "mon":
+            b += [0x1]
+        elif day == "tuesday" or day == "tue":
+            b += [0x2]
+        elif day == "wednesday" or day == "wed":
+            b += [0x4]
+        elif day == "thursday" or day == "thu":
+            b += [0x8]
+        elif day == "friday" or day == "fri":
+            b += [0x10]
+        elif day == "saturday" or day == "sat":
+            b += [0x20]
+        elif day == "sunday" or day == "sun":
+            b += [0x40]
+        elif day == "work" or day == "mon-fri":
+            b += [0x1F]
+        elif day == "weekend" or day == "sat-sun":
+            b += [0x60]
+        elif day == "week" or day == "mon-sun":
+            b += [0x7F]
```

Comment:
All these values are bitmap positions for weekdays:
monday= 00000001
saturday=00100000
work=00011111
weekend=01100000
...
The code works but guessing if there are some nice way to represent it. Maybe binary values would be more easy to guess.